src/components/NavBar/NavigationBar.tsx

```diff
@@ -108,18 +106,39 @@ const SignOutButtonContainer = styled(NavElementContainer)`
 
 const LoginDependent: React.FC<Props> = (props) => {
     const pathname = usePathname();
-
     if (pathname === "/login") {
         return <></>;
     }
-
     return <>{props.children}</>;
 };
 
 interface Props {
     children?: React.ReactNode;
 }
 
+interface RoleProps {
+    children?: React.ReactNode;
+    role: string;
+    pathname: string;
+}
+
+const RoleDependent: React.FC<RoleProps> = ({ children, role, pathname }) => {
+    const hiddenPages = role
+        ? roleToHiddenPages[role]
+        : ["/clients", "/lists", "/calendar", "/admin"];
```

Comment:
I _think_ I'd rather specify 'pages they CAN see' - if we add a new admin page, it shouldn't be 'everyone sees it unless you say otherwise'!

Reply:
Refactored